src/shipper.py

```diff
@@ -26,34 +32,61 @@ class UnknownURL(Exception):
 
 
 class LogzioShipper(object):
-    MAX_BULK_SIZE_IN_BYTES = 1 * 1024 * 1024
+    MAX_BULK_SIZE_IN_BYTES = 3 * 1024 * 1024
 
     def __init__(self, logzio_url):
-        self._size = 0
-        self._logs = []
+        self._decompress_size = 0
+        self._logs = StringIO.StringIO()
         self._logzio_url = logzio_url
+        self._logs_counter = 0
+        try:
+            self._compress = os.environ['COMPRESS'].lower() in ["yes", "true", "t", "1"]
+        except KeyError:
+            self._compress = False
+        self._writer = self._create_new_writer()
+
+    def _create_new_writer(self):
+        self._logs.truncate(0)
+        return gzip.GzipFile(mode='wb', fileobj=self._logs) if self._compress else self._logs
 
     def add(self, log):
         # type: (dict) -> None
         json_log = json.dumps(log)
-        self._logs.append(json_log)
-        self._size += sys.getsizeof(json_log)
-        self._try_to_send()
+        self._writer.write("\n" + json_log) if self._decompress_size else self._writer.write(json_log)
```

Comment:
i'm not sure how productive it will be for the short term, but if you'll wrap the writer as a function or something you can just "write" instead and not need to change this part every time you want to add a writer(let's say, we'll add a bzip writer? tcp writer?)